src/org/toml/lang/core/grammar/toml.bnf

```diff
@@ -20,21 +20,29 @@
       number='regexp:(\d|_)+'
       boolean='regexp:true|false'
       key='regexp:[0-9_\-a-zA-Z]+'
+      space='regexp:\s+'
+      date='date'
     ]
 }
 
 tomlFile ::= expression *
 
-expression ::= key_value | table
+expression ::= key_value | table | table_array
 
 table ::= table_header table_entries
 table_header ::= '[' path ']'
-path ::= key ('.' key) *
-table_entries ::= key_value *
 
-key_value ::= key '=' value
+table_array ::= table_array_header table_entries
+table_array_header ::= '[' '[' path ']' ']'
+
+key_or_str ::= key | string
+path ::= key_or_str ('.' key_or_str) *
+table_entries ::= key_value *
 
-value ::= string | number | boolean | array
-array ::= '[' (value (',' value) *)? ']'
+key_value ::= key_or_str '=' value
 
+//TODO inline_table should be on the single line
```

Comment:
I think the parser can accept multiline tables, and then annotator can highlight them as errors ([docs](http://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support/syntax_highlighting_and_error_highlighting.html)).

You can see the example of an annotator [here](https://github.com/alexeykudinkin/intellij-rust/blob/master/src/org/rust/lang/annotator/RustAnnotator.kt), and a test for it [here](https://github.com/alexeykudinkin/intellij-rust/blob/master/test/org/rust/lang/RustAnnotatorTest.kt) and [here](https://github.com/alexeykudinkin/intellij-rust/blob/master/testData/annotator/attributes.rs)